app/pages/events/activity_delete.php

```diff
@@ -0,0 +1,37 @@
+<?php
+restrict_access(Access::$ADD_EVENTS);
+
+$activity_id = get_route_param("activity_id");
+$event_id = get_route_param("event_id");
+$activity = em()->find(Activity::class, $activity_id);
+if (!$activity) {
+    echo "the activity of id $activity_id doesn't exist";
```

Comment:
Prefer using formatted errors like Toast::error() or force_404().

Reply:
Same comment around copy-pasting. Maybe it is time to extract some parts of this confirmation page :D